classes/query/Repository.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+/**
+ * @file classes/query/Repository.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Repository
+ *
+ * @see Query
+ *
+ * @brief Operations for retrieving and modifying Query objects.
+ */
+
+namespace PKP\query;
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\notification\NotificationManager;
+use APP\submission\Submission;
+use Illuminate\Support\Facades\Mail;
+use PKP\db\DAORegistry;
+use PKP\mail\Mailable;
+use PKP\note\Note;
+use PKP\notification\Notification;
+use PKP\notification\NotificationSubscriptionSettingsDAO;
+use PKP\security\Role;
+use PKP\stageAssignment\StageAssignment;
+use PKP\user\User;
+
+class Repository
+{
+    /**
+     * Retrieve a count of all open queries totalled by stage
+     *
+     * @param int[] $participantIds Only include queries with these participants
+     *
+     * @return array<int,int> [int $stageId => int $count]
+     */
+    public function countOpenPerStage(int $submissionId, ?array $participantIds = null): array
+    {
+        $counts = Query::withAssoc(Application::ASSOC_TYPE_SUBMISSION, $submissionId)
+            ->when($participantIds !== null, function ($q) use ($participantIds) {
+                $q->withUserIds($participantIds);
+            })
+            ->withClosed(false)
+            ->selectRaw('stage_id, COUNT(stage_id) as count')
+            ->groupBy('stage_id')
+            ->get()
+            ->mapWithKeys(fn ($row, $key) => [$row->stage_id => $row->count])
+            ->toArray();
+
+        return collect(Application::get()->getApplicationStages())
+            ->mapWithKeys(fn ($stageId, $key) => [$stageId => $counts[$stageId] ?? 0])
+            ->toArray();
+    }
+
+    /**
+     * Sequentially renumber queries in their sequence order.
+     *
+     * @param int $assocType Application::ASSOC_TYPE_...
+     * @param int $assocId Assoc ID per assocType
+     */
+    public function resequence($assocType, $assocId): void
+    {
+        $result = Query::withAssoc($assocType, $assocId)
+            ->orderBy('seq')
+            ->get();
+
+        $result->each(function ($item, $key = 1) {
+            Query::find($item->queryId)
+                ->update(['seq' => $key]);
```

Comment:
I'm giving this a try, but run into issues with `upsert` here, e.g.:  `Column not found: 1054 Unknown column 'assocId' in 'field list'`.

This may be due to the HasCamelCasing trait, where `upsert` is expecting the exact field name.

Reply:
@kaitlinnewson thanks for giving the upsert approach a try. I see the issue with `upsert` and the `HasCamelCasing`. In this case @touhidurabir suggestion will work since we already have the collection from `get()`.  you can update the sequence directly within the` each()` loop without needing to use `find()` again. this will avoid the extra lookup. 😊